### PR TITLE
fix undeclared variables or wrong variable names (/mod) part 2

### PR DESCRIPTION
**Pull request by rabuzarus**
Some variables were not declared. Some variables were not available. And some variables were did have a typo.
This PR fixes this issue for the files in the /mod directory.

I couldn't solve following "undeclared variable" issues in /mod:
1.) /mod/editpost.php
L66 - https://github.com/friendica/friendica/blob/develop/mod/editpost.php#L64
How do we get the `$group` variable? (does also affect L143)

2.) /mod/follow.php
L47 - https://github.com/friendica/friendica/blob/develop/mod/follow.php#L47
`$contact_id` isn't declared. But I don't know the intended beavior so I couldn't fix it.

3.) /mod/poke.php.
L144 - https://github.com/friendica/friendica/blob/develop/mod/poke.php#L144
We have `$item_id` but what is `$post_id` and from where should we get it?

Note:
By adding missing variables the code should be fixed but it could have an impact on how the code will work. Keep this in mind when reviewing the PR.

### CODE REVIEW

File: mod/register.php [hunk 1]
```diff
@@ -58,6 +58,7 @@ function register_post(App $a)
 			break;
 	}
 
+	$netpublish = (((x($_POST, 'profile_publish_reg')) &intval($_POST['profile_publish_reg']) = 1) ? 1 : 0);
```
Comment (MrPetovan) [inline mod/register.php]: This is wrong (single equal sign) and could be shortened to:
```php
$netpublish = !empty($_POST['profile_publish_reg']);
```

File: mod/viewcontacts.php [hunk 2]
```diff
@@ -46,6 +46,8 @@ function viewcontacts_content(App $a)
 		return;
 	}
 
+	$is_owner = ((local_user() && ($a->profile['profile_uid'] == local_user())) ? true : false);
```
Comment (MrPetovan) [inline mod/viewcontacts.php]: Can be shortened to 
```php
$is_owner = $a->profile['profile_uid'] == local_user();
```

**Comment by MrPetovan** [comment]
When I did such refactoring work, I added `@todo` comments where I couldn't guess a variable's origin. It makes the issues pop up in the diff and leaves a mark for future maintainers to do something about it.

**Comment by rabuzarus** [comment]
Yes but I would go through all undeclared variables. So that we have at the end no undeclared variables left. I would love to see a solution in the next days for the remaining undeclared variables (btw. deleting is also an option :-)   ).
I would raise the discussion now. Otherwise it will be in the code for years.

**Comment by MrPetovan** [comment]
Both aren't incompatible. In my previous PRs, `@todo` elements allowed @annando to pitch in without me having to list them in the PR post itself like you did. I ended up removing them in the same PR following @annando's suggestions.

**Comment by rabuzarus** [comment]
OK, I'll do it next time

**Comment by MrPetovan** [comment]
I can merge right now, do you want me to wait for other people to take a look at it?

**Comment by rabuzarus** [comment]
please wait. Maybe @annando does have any idea about the missing variables :-)

**Comment by MrPetovan** [comment]
Related to #4176

**Comment by annando** [comment]
Since I'm too lazy and still not totally recovered from the cold: What am I supposed to do?

**Comment by rabuzarus** [comment]
Please read my questions (1, 2, 3) from the PR description. I would an advice how to deal with this issues. Problem is that the mentioned variables are undeclared so they don't execute code like expected.

File: mod/tagger.php [hunk 3]
```diff
@@ -72,7 +72,7 @@ function tagger_content(App $a) {
 	$targettype = (($item['resource-id']) ? ACTIVITY_OBJ_IMAGE : ACTIVITY_OBJ_NOTE );
 
 	$link = xmlify('<link rel="alternate" type="text/html" href="'
-		. System::baseUrl() . '/display/' . $r['nickname'] . '/' . $item['id'] . '" />' . "\n") ;
+		. System::baseUrl() . '/display/' . $owner_nick . '/' . $item['id'] . '" />' . "\n") ;
```
Comment (MrPetovan) [inline mod/tagger.php]: It seems as `$owner_nick` may not be always initialized.

**Comment by annando** [comment]
Concerning ```mod/editpost.php```: The lockstate variable should depend on the "item" array (and there the "allow_cid", "deny_cid", "allow_gid" and "deny_gid" variables and not like it is done now.

```mod/follow.php```: The lines from 44 does make only partly sense. AFAIK these lines are only executed if no contact was created. So lines 46 to 48 should be removed and the message in line 44 should be renamed.

```mod/poke.php```: Just remove the line 144, it is obsolete, I think.

**Comment by rabuzarus** [comment]
What do we do with the `$bang` template variable? Have a look at https://github.com/friendica/friendica/blob/develop/mod/editpost.php#L143
I have no idea what this template variable is for.

**Comment by MrPetovan** [comment]
It looks like it is for forum mentions, but I don't know the jot template enough.

**Comment by rabuzarus** [comment]
I would guess `$bang` is just a visual sign (exclamation mark) if the post rights are restricted.

**Comment by rabuzarus** [comment]
I would say all tasks are complete and it can be merged

**Pull request merged**